aquarius/app/dao.py

```diff
@@ -121,4 +123,30 @@ def run_es_query(self, data):
         for x in page["hits"]["hits"]:
             object_list.append(x["_source"])
 
-        return object_list, page["hits"]["total"]
+        if not with_metadata:
+            return object_list, page["hits"]["total"]
+
+        body = {
+            "size": 0,
+            "query": query,
+            "aggs": {
+                "licenses": {
+                    "terms": {"field": "service.attributes.main.license.keyword"}
+                },
+                "tags": {
+                    "terms": {
+                        "field": "service.attributes.additionalInformation.tags.keyword"
+                    }
+                },
+            },
+        }
+        logging.info(f"running metadata query: {body}")
+        metadata = self.oceandb.driver.es.search(
+            index=self.oceandb.driver.db_index, body=body
+        )
+        metadata = {
+            key: rename_metadata_keys(value["buckets"])
+            for key, value in metadata["aggregations"].items()
+        }
+
+        return object_list, page["hits"]["total"], metadata
```

Comment:
it may be much cleaner to leave this function as is and add another function to handle the extra metadata.